src/pages/task/taskTabs/logs/LogTypes.tsx

```diff
@@ -174,32 +172,20 @@ const useRenderBody: React.VFC<{
   loading: boolean;
   error: ApolloError;
   data: (TaskEventLogEntryType | LogMessageType)[];
-  currentLog: LogTypes;
   LogContainer?: React.VFC<{ children: React.ReactNode }>;
-  htmlLink: string;
-  rawLink: string;
-  lobsterLink: string;
+  setNoLogs: (noLogs: boolean) => void;
 }> = ({
   loading,
   error,
   data,
-  currentLog,
-  rawLink,
-  htmlLink,
-  lobsterLink,
   LogContainer = ({ children }) => <StyledPre>{children}</StyledPre>,
+  setNoLogs,
 }) => {
-  const taskAnalytics = useTaskAnalytics();
-  const updateQueryParams = useUpdateURLQueryParams();
   const noLogs = !!((error && !data) || !data.length);
-  const onChangeLog = (event: React.ChangeEvent<HTMLInputElement>): void => {
-    const nextLogType = event.target.value as LogTypes;
-    updateQueryParams({ [QueryParams.LogType]: nextLogType });
-    taskAnalytics.sendEvent({
-      name: "Select Logs Type",
-      logsType: nextLogType,
-    });
-  };
+  // Update the value of noLogs in the parent component.
+  useEffect(() => {
+    setNoLogs(noLogs);
+  }, [setNoLogs, noLogs]);
 
   let body = null;
```

Comment:
[nit] We may as well just return each of these JSX chunks rather than assign them to `body`